### ISSUE TITLE
Issue #2 - Database Schema

### ISSUE THREAD

**Issue by keginzburg** [adedushi/sike#3]
# Database Schema
- [X] Contains correct datatypes
- [X] Contains appropriate constraints/details
  - [X] primary key
  - [X] not null
  - [X] unique
  - [X] indexed
  - [X] foreign key
- [X] Contains bullet points after the table that state which foreign keys will reference to which table, or references to the associations which will be made
 - [X] foreign key and table name are lowercased, snake_cased and `back_ticked`
- [X] Correctly formatted
  - [X] schema is written in a table format
  - [X] the table's name are lowercased, snake_cased and `back_ticked`
  - [X] the table header column names are bolded
  - [X] columns names are lowercased and snaked_cased and `back_ticked`

## Comments
1. Creating a separate table for the `Options` of your Shoe Stocks and Sizes is creative. Let me know if you'd like to discuss this down the line.
2. Is there a reason why your `options` table has an `index on session_token`?
3. Besides their `name`, `description`, and `price`, does Nike provide any additional info for Shoes not covered by our `options` table? The products are this clone's selling point, so the more information for them in their table, the better.
4. When a user favorites a shoe on this application, do they favorite the shoe or the stock and size? If it's all of the above, we can scale this down and make our `favorites` table just connect the `users` and `products` table.
5. For our purposes, I think we should nix the `session_token` from the `carts` table. While on most sites, we can connect our cart to our browser cookies, we're going to opt for only letting a user create a cart if logged in. With that, we can just pull their cart from the database using their `user_id`. Let me know if you'd like to discuss.
6. Great information within the `Reviews` table.

**Comment by adedushi** [comment]
1. I certainly see the value of having a few separate tables for product attributes, partially inspired by this  [stack overflow answer](https://stackoverflow.com/a/31750974). For simplicity, I could remove the table altogether but I would like to discuss it beforehand.
2. I removed the `index on session_token` from `options`
3. I added a row for `gender` to the `products` table. I can add more depending on how comment 1 is resolved. Photos will also fill up much of the page
4. On Nike's website, users can favorite a specific size or product. I can scale it down and only allow favorites for `products`
5. I'm open to removing `session_token` from carts if we decide only logged in users can have carts
